### ISSUE TITLE
Window Pop up after typing scrcpy in cmd but gets closed automatically in 2 sec.

### ISSUE THREAD

**Issue by 9374020304** [Genymobile/scrcpy#1600]
- [ ] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [ ] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: [Windows 10 64bit]
 - scrcpy version: [e.g. 1.14]
 - installation method: [e.g. manual build, apt, snap, brew, Windows release...]
 - device model: Samsung Galaxy S10 Lite
 - Android version: [10]

**Describe the bug**
I just want to record my gameplay with scrcpy.
I already have adb and scrcpy on my pc and also started USB Deugging on My Device.
adb device command also shows my device.
but when i type scrcpy command a window just pop up showing device name on it and it just get closed automatically.
is there any solution for that i have been working on this thing for last 2 days.
There is no error the window for mirroring pop up and get closed automatically in 2 sec.
Uploading a screenshot with the command i used.
![Error](https://user-images.githubusercontent.com/68462844/87849472-a6952380-c906-11ea-81d0-7ba365d44b84.jpg)

**Comment by rom1v** [comment]
I guess adb is getting disconnected. Open a separate console with `adb logcat` and keep it open. If it terminates at the same tile as scrcpy, then it's adb which disconnect.

In that case, try to use another USB cable or port (see #281 #283).

**Comment by rom1v** [comment]
Otherwise, there is probably an error printed in `adb logcat`, please post it.

**Comment by 9374020304** [comment]
i started another console with adb logcat and tried using scrcpy in another window first it shows some error i am attaching a file on that and in another try the cursor just keep blinking i need a solution asap please do something.
![error 1](https://user-images.githubusercontent.com/68462844/87850131-78661280-c90b-11ea-8b55-8ebd817a4062.jpg)

**Comment by 9374020304** [comment]
> Otherwise, there is probably an error printed in `adb logcat`, please post it.

Hoping a Reply ASAP!!!

**Comment by 9374020304** [comment]
Can i get a answer please i am working on it for last 3 days atleast in tomorrow morning it gets connected and i was mirroring for 10 mins but then again it gets connected and after 1 sec it just close the app i need a solution please that is my last hope for mirroring.

> Otherwise, there is probably an error printed in `adb logcat`, please post it.

**Comment by 9374020304** [comment]
![error](https://user-images.githubusercontent.com/68462844/87925210-94012280-ca9d-11ea-867e-2138bac918b6.jpg)
i am getting this error after i use the command
scrcpy -Vdebug
Please Help!!

**Comment by rom1v** [comment]
Tl dump the logs:

```bash
adb logcat -c  # clean the logs
scrcpy
# once it is closed
adb logcat -d > logcat.txt
```

Then post your `logcat.txt`.

**Comment by 9374020304** [comment]
[logcat.txt](https://github.com/Genymobile/scrcpy/files/4947224/logcat.txt)
Here it is.

**Comment by rom1v** [comment]
I don't find any error specific to scrcpy. A lot of errors related to CamX...

> i was mirroring for 10 mins but then again it gets connected and after 1 sec it just close the app

So it's probably a crash in your device system. If you reboot, does it work again?

Does `adb shell screenrecord /sdcard/file.mp4` work?
What about `scrcpy -m800`?

**Comment by 9374020304** [comment]
> I don't find any error specific to scrcpy. A lot of errors related to CamX...
> 
> > i was mirroring for 10 mins but then again it gets connected and after 1 sec it just close the app
> 
> So it's probably a crash in your device system. If you reboot, does it work again?
> 
> Does `adb shell screenrecord /sdcard/file.mp4` work?
> What about `scrcpy -m800`?

I have already tried 3 or 4 devices i dont think there is any error in my device.

**Comment by rom1v** [comment]
> I have already tried 3 or 4 devices

Did you try with another cable/USB port? (#281 #283)

**Comment by 9374020304** [comment]
yeah i tried it, the window just open and get closed in 1 sec

**Comment by rom1v** [comment]
If this happens to several (different) devices, the problem is on the computer-side. Could you try with another computer/OS?

**Comment by 9374020304** [comment]
yup i just tried using another pc and it works just fine. So that is clear the problem is with my pc.
Is there any solution for that??

**Comment by rom1v** [comment]
It might be due to insufficient USB power. Or maybe bad USB drivers...

Try with another OS (run a live CD of some Linux distrib to test).

**Comment by 9374020304** [comment]
Should i try formatting my pc?

**Comment by 9374020304** [comment]
is it gonna help?

**Comment by 9374020304** [comment]
can i get your email or something so we can personally i really want to run scrcpy i can even format my whole laptop for this to work u can imagine how important it is for me to get it working.
i have to invest in some capture card if i dont get any solution from you please help me with that.
i know that the problem is in the laptop so should i format it?

**Comment by rom1v** [comment]
Download a live cd of Ubuntu for example, and run it. No need to format.

**Comment by 9374020304** [comment]
ok thank you i will try that and contact again.

**Comment by 9374020304** [comment]
but i have one more question if it will work with ubuntu how can i make it work on my windows?
i want to run it on my windows and record it.

**Comment by 9374020304** [comment]
Can u answer?

**Comment by 9374020304** [comment]
Yup working just fine in the ubuntu latest version but there is some problem in win 10 i think.
i tried playing around in ubuntu live cd but it lags so much and frame drops continuously so i think i have to invest in a capture card now. Thank You so much for all the help and support.

**Comment by rom1v** [comment]
(It lags probably because it's a livecd without all the firmware/drivers. It should work with a real install.)

**Comment by 9374020304** [comment]
but now how can i get it working on my windows?

**Comment by rom1v** [comment]
I don't know. My guess is that disconnections are due to a problem with your USB drivers or USB power. If it works from Ubuntu, then it's probably your drivers or some Windows problem.

(Could you confirm that `adb logcat` in a separate terminal terminates at the same time as scrcpy is unexpectedly disconnected?)

**Comment by rom1v** [comment]
Does it work better with v1.9 or v1.10 instead of v1.14?

**Comment by rom1v** [comment]
Did you try to reduce the definition? https://github.com/Genymobile/scrcpy/issues/283#issuecomment-672939254
